### PR TITLE
feat: add retry metadata to fsm decl

### PR DESCRIPTION
**Pull request by matt2e**
Changes:
- Extract retries metadata declared for a FSM. The idea is that this will be the default retry policy for each transition, but that is out of scope for this PR.
- Traversing the ast graph now provides a stack of nodes rather than just the current node (so we can derive info based on parent nodes, rather than maintaining that state with special cases)
    - removed `currentVerb` from `parserContext` as this special case is no longer needed

Add retries to a FSM like this:
```go
// The payment FSM.
//
//ftl:retry 10 5s 10m
var paymentFSM = ftl.FSM("payment",
	ftl.Start(Created),
	ftl.Start(Paid),
	ftl.Transition(Created, Paid),
	ftl.Transition(Created, Failed),
	ftl.Transition(Paid, Completed),
)
```
In the schema it looks like this:
```
fsm payment
   +retry 10 5s 10m
{
   start fsm.created
   start fsm.paid
   transition fsm.created to fsm.paid
   transition fsm.created to fsm.failed
   transition fsm.paid to fsm.completed
}

### CODE REVIEW

**Comment by alecthomas** [comment]
> In goast, there is no relationship between ftl.FSM(...) and the comment group, so this PR makes a map of line number to comment group node, which we can then look up when we find the FSM function call.

You should be able to find variables declarations of the FSM type, then get the comments from the variable no?

**Comment by alecthomas** [comment]
> ## How should retries for FSMs show up in schema?

I like this syntax 👍

**Comment by matt2e** [comment]
> > In goast, there is no relationship between ftl.FSM(...) and the comment group, so this PR makes a map of line number to comment group node, which we can then look up when we find the FSM function call.
> 
> You should be able to find variable declarations of the FSM type, then get the comments from the variable no?

Ah you're right, the variable declaration holds the documentation.
I've made `Visit()` provide a stack which makes this a lot easier to do. And it removes the need for `parserContext` to track `activeVerb`

File: internal/goast/visitor.go [hunk 1]
```diff
@@ -3,561 +3,369 @@ package goast
...
 // The visitor function can call next() to continue traversal.
 //
 // Note that this is based on a direct copy of ast.Walk.
-func Visit(node Node, v VisitorFunc) error { //nolint:maintidx
-	return v(node, func() error {
+func Visit(node goast.Node, v VisitorFunc) error {
+	return visitStack([]goast.Node{node}, v)
+}
+
+func visitStack(stack []goast.Node, v VisitorFunc) error { //nolint:maintidx
```
Comment (matt2e) [inline internal/goast/visitor.go:20]: regex did the dirty work here, no RSI was harmed

**Pull request merged**